.github/dependabot.yml

```diff
@@ -1,19 +0,0 @@
-# To get started with Dependabot version updates, you'll need to specify which
-# package ecosystems to update and where the package manifests are located.
-# Please see the documentation for all configuration options:
-# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
-
-version: 2
-updates:
-  - package-ecosystem: "pip" # See documentation for possible values
-    directory: "/" # Location of package manifests
-    schedule:
-      interval: "daily"
-    # Add assignees
-    assignees:
-      - "math-fehr"
-      - "georgebisbas"
-    commit-message:
-      prefix: "pip prod"
-      prefix-development: "pip dev"
-      include: "scope"
```

Comment:
Could we actually keep dependabot, though maybe making the schedule to "weekly" instead?

I feel that `update-bot` is not as strong as dependabot, as it will not update the actual version, it will just update the minor ones (and miss security updates).
Having both at the same time is probably what we want here.

Reply:
> tl;dr, I've just now caught up with you in realising it might be possible to use Dependabot in this case despite the docs saying its not supported. I'll try to make changes to go back to just using Dependabot and see if something breaks


Long chain of thought to justify to myself this when I come back to it tomorrow:

I've just had a further stare at this and am wondering if Dependabot might just work for `uv` despite not being explicitly supported?

On modern python projects the requirements are often loosely defined in the `pyproject.toml` file (e.g. [`package = ^1.2.1`](https://github.com/Textualize/textual/blob/main/pyproject.toml#L59)), then the specific versions set in a lock file such as `Pipenv.lock`/`poetry.lock`/`uv.lock` - then Dependabot acts [on this lock file](https://github.com/xdslproject/xdsl/pull/3366). Since `uv` isn't a supported lock file type, Dependabot can't be used like this.

However, I've just now realised that since this project is pinning dependency versions in `pyproject.toml` (as `pip` doesn't have a lockfile), Dependabot acts on the `pyproject.toml` [directly](https://github.com/xdslproject/xdsl/pull/3366) -- which I hadn't seen before so didn't think to check. As such if we either: don't commit the `uv` lock file; or use Dependabot in conjunction with update-bot to bump the lock file, it should all work correctly

In future, it might be worth considering using a lock file instead of pinning dependencies in `pyproject.toml` -- but for now I think it saves us from this problem so we can just leave it.